### PR TITLE
Propagate failures in pandas integration tests and Skip failing tests

### PR DESCRIPTION
**Pull request by Matt711**
## Description

This PR ensures that the integration tests fail in any one of the test modules fails. It also skips of xfails any tests that are not currently passing. Finally, it fixes one incorrect use of `rng.random`.

Some of the change were originally made in  #17489
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by Matt711** [comment]
For the reviewer, I'll follow-up this PR to fix failing tests. xref #17490

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py [hunk 1]
```diff
@@ -113,6 +113,9 @@ def test_with_external_memory(
     return predt
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:118]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_tensorflow.py [hunk 2]
```diff
@@ -271,6 +271,7 @@ def call(self, values):
         return tf.concat(values, axis=-1)
 
 
+@pytest.mark.xfail(reason="ValueError: Invalid dtype: object")
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_tensorflow.py:274]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_seaborn.py [hunk 3]
```diff
@@ -54,6 +54,9 @@ def test_scatter(df):
     return ax
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_seaborn.py:59]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py [hunk 4]
```diff
@@ -116,6 +116,9 @@ def test_torch_train(data):
     return model(test_x1, test_x2)
 
 
+@pytest.mark.xfail(
+    reason="AssertionError: The values for attribute 'device' do not match: cpu != cuda:0."
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py:121]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_numpy.py [hunk 5]
```diff
@@ -37,6 +37,9 @@ def test_numpy_dot(df):
     return np.dot(df, df.T)
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_numpy.py:42]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_matplotlib.py [hunk 6]
```diff
@@ -33,6 +33,9 @@ def assert_plots_equal(expect, got):
 pytestmark = pytest.mark.assert_eq(fn=assert_plots_equal)
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_matplotlib.py:38]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_holoviews.py [hunk 7]
```diff
@@ -71,9 +71,15 @@ def test_holoviews_heatmap(df):
...
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
 def test_holoviews_histogram(df):
     return get_plot_info(hv.Histogram(df.values))
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_holoviews.py:83]:
```suggestion
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_holoviews.py [hunk 8]
```diff
@@ -71,9 +71,15 @@ def test_holoviews_heatmap(df):
     )
 
 
+@pytest.mark.xfail(
+    reason="AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"
+)
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_holoviews.py:76]:
```suggestion
```

**Review by vyasr — COMMENTED** [review]
Is the plan to merge this after we add xfails? Sorry I think I've lost track of the various discussions we've had, I know I've asked this a couple of times alrady.

**Comment by Matt711** [comment]
> Is the plan to merge this after we add xfails? Sorry I think I've lost track of the various discussions we've had, I know I've asked this a couple of times already.

Yes that is still the plan. A few of the tests that are failing with `"AttributeError: 'ndarray' object has no attribute '_fsproxy_wrapped'"` are giving me some weird problems. They fail when run normally and XPASS when I add `@pytest.mark.xfail`. I'm going to look at these failures more closely tomorrow. And my apologies, I meant to update you earlier today.

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py [hunk 9]
```diff
@@ -3,7 +3,13 @@
...
 import pandas as pd
 import pytest
-from catboost import CatBoostClassifier, CatBoostRegressor, Pool
+
+try:
+    from catboost import CatBoostClassifier, CatBoostRegressor, Pool
+except Exception:
+    pytest.skip(
+        "ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject"
+    )
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py:12]:
```suggestion
    pytest.skip(
        reason="ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject", allow_module_level=True
    )
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py [hunk 10]
```diff
@@ -3,7 +3,13 @@
 import numpy as np
 import pandas as pd
 import pytest
-from catboost import CatBoostClassifier, CatBoostRegressor, Pool
+
+try:
+    from catboost import CatBoostClassifier, CatBoostRegressor, Pool
+except Exception:
+    pytest.skip(
+        reason="ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject", allow_module_level=True
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py:11]:
```suggestion
        reason="ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject",
        allow_module_level=True,
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py [hunk 11]
```diff
@@ -1,129 +0,0 @@
-# Copyright (c) 2024, NVIDIA CORPORATION.
```
[the remark this answers is not in the document]
Comment (Matt711) [reply python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py:1]: This is probably due to ABI incompatibility between the version of NumPy that catboost is compiled against and the version of of NumPy installed in test CI job.

**Comment by vyasr** [comment]
Why delete catboost? It looks like the failure is a runtime issue and not like a seg fault, so can't you just mark the test as a skip?

**Review by vyasr — APPROVED** [review]

File: .github/workflows/pr.yaml [hunk 12]
```diff
@@ -40,6 +40,7 @@ jobs:
       - pandas-tests
       - pandas-tests-diff
       - telemetry-setup
+      - third-party-integration-tests-cudf-pandas
```
Comment (Matt711) [inline .github/workflows/pr.yaml:43]:
```suggestion
```

File: .github/workflows/pr.yaml [hunk 13]
```diff
@@ -325,6 +326,17 @@ jobs:
...
+    needs: wheel-build-cudf
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/custom-job.yaml@branch-24.10
+    with:
+      build_type: pull-request
+      node_type: "gpu-v100-latest-1"
+      arch: "amd64"
+      container_image: "rapidsai/ci-conda:latest"
+      run_script: |
+        ci/cudf_pandas_scripts/third-party-integration/test.sh python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml
```
Comment (Matt711) [inline .github/workflows/pr.yaml:339]:
```suggestion
```

**Pull request merged**